build.gradle

```diff
@@ -86,9 +89,25 @@ allprojects {
 
 dependencies {
     compileOnly 'com.google.code.findbugs:jsr305:3.0.2'
+    testImplementation project(":")
```

Comment:
Why is this needed?

Reply:
We change the wiring of the test task for all runs:
Normally, the test task uses:
- output of the `main` source set
- output of the `test` source set
- testRuntimeClasspath

It is impossible to use dependency substitution for replacing the output of the main source set. So we remove the output of the main source set from the test classpath and replace it by a dependency on `project(':')` instead. That is why we need to add the dependency on project(':') here.

Reply:
I added some comments explaining what we do there.